NorthstarDLL/squirrelnativeautobind.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+#include <vector>
+
+typedef void (*SqAutoBindFunc)(void);
+
+class SquirrelAutoBindContainer
+{
+  public:
+	std::vector<SqAutoBindFunc> clientSqAutoBindFuncs;
+	std::vector<SqAutoBindFunc> serverSqAutoBindFuncs;
+
+	SquirrelAutoBindContainer()
+	{
+		clientSqAutoBindFuncs = {};
+		serverSqAutoBindFuncs = {};
+	}
+};
+
+extern SquirrelAutoBindContainer* g_pSqAutoBindContainer;
+
+class __squirrelautobind;
+
+#define ADD_SQUIRREL_FUNC(returnType, funcName, argTypes, helpText, runOnContext)                                                          \
```

Comment:
this could probably be shorter, you'd honestly be good with just SQFUNC or something as your macro name